tests/components/dsmr/test_sensor.py

```diff
@@ -47,6 +48,36 @@ async def connection_factory(*args, **kwargs):
     return connection_factory, transport, protocol
 
 
+async def test_setup_platform(hass, mock_connection_factory):
+    """Test setup of platform."""
+    async_add_entities = MagicMock()
+
+    entry_data = {
+        "port": "/dev/ttyUSB0",
+        "dsmr_version": "2.2",
+        "precision": 4,
+        "reconnect_interval": 30,
+    }
+
+    result = await sensor.async_setup_platform(
```

Comment:
we dont want to call integration methods here. you have to setup the platform using HA methods.

example: https://github.com/home-assistant/core/blob/dev/tests/components/sonarr/test_sensor.py#L34

Reply:
Ok, copied this code from zwave test code, but apparently it's a bad example. Changed it.